src/blockrelay/blockrelay_common.h

```diff
@@ -54,6 +55,10 @@ class ThinTypeRelay
     std::atomic<int32_t> nGraphenePeers{0};
     std::atomic<int32_t> nCompactBlockPeers{0};
 
+public:
+    // blocks still in flight sent by the sender.
+    std::map<NodeId, std::shared_ptr<CGrapheneBlock> > mapGrapheneSentBlocks GUARDED_BY(cs_graphene_sender);
+
```

Comment:
does this really need to be public...if you're only accessing it one time publicly for deleting an entry then you could just have a method to for that one need.

Reply:
I made it private and added a Clear method.

Reply:
looks the same to me, did you push your change?

Reply:
actually looks like you added the clear but didn't make the map private.

Reply:
Ah yes, sorry about that. I just pushed a fix. It turned out to be necessary to add a setter and getter for mapGrapheneSentBlocks since it is also accessed in graphene.cpp.

Reply:
Come to think of it isn't this data already tracked and stored in mapblocksreconstruct...each block of CBlockThinRelay has a shared pointer to either of the thintype blocks and would have stored the graphene block or partially reconstructed graphene block there already basically making this map redundant?

Reply:
The data in `mapGrapheneSentBlocks` corresponds to block sent by the sender, whereas `mapBlocksReconstruct` are blocks being reconstructed by the receiver.

Reply:
Unless I missed something, I also don't see anywhere where this mapGrapheneSentBlocks is being used?

Reply:
never mind, I found it.